src/components/carbon-provider/carbon-provider.pw.tsx

```diff
@@ -0,0 +1,341 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+import { HooksConfig } from "../../../playwright";
+
+import {
+  SageTheme as SageThemeStory,
+  MintTheme as MintThemeStory,
+} from "../../../src/components/carbon-provider/carbon-provider.stories";
+import { AllThemes as AllThemesStory } from "../../../src/components/carbon-provider/carbon-provider-test.stories";
+import Button from "../../../src/components/button";
+import Link from "../../../src/components/link";
+import Loader from "../../../src/components/loader";
+import LoaderBar from "../../../src/components/loader-bar";
+import MultiActionButton from "../../../src/components/multi-action-button";
+import Pill from "../../../src/components/pill";
+import SplitButton from "../../../src/components/split-button";
+import StepSequenceItem from "../../../src/components/step-sequence/step-sequence-item";
+import { Tabs, Tab } from "../../../src/components/tabs";
+
+import { buttonDataComponent } from "../../../playwright/components/button/index";
+import {
+  linkComponent,
+  loaderBarComponent,
+  loaderComponent,
+} from "../../../playwright/components/themes/index";
+import { pillPreview } from "../../../playwright/components/pill/index";
+import { stepSequenceDataComponentItem } from "../../../playwright/components/step-sequence/index";
+import { multiActionButtonComponent } from "../../../playwright/components/multi-action-button/index";
+import { checkAccessibility } from "../../../playwright/support/helper";
+
+const themes = ["mint", "aegean", "noMountedTheme", "sage"];
+
+const buildTestArray = (array: string[]) => {
+  return themes.map((theme, i) => [theme, array[i]]);
+};
+
+const buildTestDataWithTwoArrays = (
+  firstArray: string[],
+  secondArray: string[]
+) => {
+  return themes.map((theme, i) => [theme, firstArray[i], secondArray[i]]);
+};
+
+const commonColors = [
+  "rgb(0, 125, 90)",
+  "rgb(0, 115, 194)",
+  "rgb(0, 130, 0)",
+  "rgb(0, 126, 69)",
+];
+
+const commonColorsOnHover = [
+  "rgb(0, 96, 70)",
+  "rgb(0, 92, 154)",
+  "rgb(0, 99, 0)",
+  "rgb(0, 103, 56)",
+];
+
+const multiButtonOnHover = [
+  "rgb(0, 64, 46)",
+  "rgb(0, 68, 114)",
+  "rgb(0, 77, 42)",
+  "rgb(0, 77, 42)",
+];
+
+const stepSequenceColors = [
+  "rgb(0, 125, 90)",
+  "rgb(0, 115, 194)",
+  "rgb(0, 138, 33)",
+  "rgb(0, 138, 33)",
+];
+
+const loaderBarColors = [
+  "rgb(179, 227, 214)",
+  "rgb(179, 214, 239)",
+  "rgb(179, 224, 179)",
+  "rgb(179, 217, 200)",
+];
+
+buildTestArray(commonColors).forEach(([theme, color]) => {
+  test.describe(
+    `Carbon Provider should render components with ${theme} theme`,
```

Comment:
**suggestion** (non-blocking): I personally would just call the description "Carbon Provider"...